viadot/sources/databricks.py

```diff
@@ -0,0 +1,436 @@
+from numpy import source
+from viadot.config import local_config
+from viadot.sources.base import Source
+from delta.tables import *
+import pandas as pd
+from pyspark import SparkConf, SparkContext
+import pyspark.sql.dataframe as spark
+from pyspark.sql.dataframe import DataFrame
+from prefect.utilities import logging
+from typing import Literal, Union
+from typing import Any, Dict
+
+import viadot.utils
+
+
+logger = logging.get_logger(__name__)
+
+
+class Databricks(Source):
+    """
+    A class for pulling and manipulating data on Databricks.
+
+    Documentation for Databricks is located at: https://docs.microsoft.com/en-us/azure/databricks/
+
+    Parameters
+    ----------
+    env : str, optional
+        The name of the Databricks environment to use. by default "DEV"
+
+    credentials : Dict[str, Any], optional
+        Credentials containing Databricks connection configuration and the following credentials:
+            - spark.databricks.service.address
+            - spark.databricks.service.token
+            - spark.databricks.service.clusterId
+            - spark.databricks.service.orgId
+            - spark.databricks.service.port
+    """
+
+    env = "DEV"
+    session = None
+    DEFAULT_SCHEMA = "default"
+
+    def __init__(
+        self, env: str = "DEV", credentials: Dict[str, Any] = None, *args, **kwargs
+    ):
+        self.env = env
+
+        self.credentials = credentials or local_config.get("DATABRICKS", {}).get(env)
+
+        self.connect()
+
+    def _get_spark_session(self, env: str = "DEV"):
```

Comment:
This method created a new spark session, reads config, adjusts it and returns a `new_spark`. Shouldn't it be named more like `_create_spark_session` than `_get_spark_session`?

Moreover, we could name `new_config` just as `config`, as there is no *old* config or any other local config variable.